source/features/merge-status.tsx

```diff
@@ -0,0 +1,64 @@
+import './merge-status.css';
+import select from 'select-dom';
+import React from 'dom-chef';
+import * as api from '../libs/api';
+import {getOwnerAndRepo} from '../libs/utils';
+import features from '../libs/features';
+
+const CONFLICTING = 'CONFLICTING';
+
+function buildQuery(
+	ownerName: string,
+	repoName: string,
+	prs: string[],
+): string {
+	return `
+		query {
+			repository(owner: "${ownerName}", name: "${repoName}") {
+				${prs.map(
+		(pr: string) => `
+					${pr}: pullRequest(number: ${pr.replace('issue_', '')}) {
+						mergeable
+					}
+				`,
+	)}
+			}
+		}
+	`;
+}
+
+async function init(): Promise<false | void> {
+	const elements = select.all('.js-issue-row');
+	if (elements.length === 0) {
+		return false;
+	}
+
+	const {ownerName, repoName} = getOwnerAndRepo();
+
+	const query = buildQuery(ownerName, repoName, elements.map(e => e.id));
+
+	const data = await api.v4(query);
+
+	for (const pr of elements) {
+		if (data.repository[pr.id].mergeable === CONFLICTING) {
+			select('.d-inline-block.mr-1 > .commit-build-statuses', pr)!.before(
+				<span
```

Comment:
Can this become a link? I know that some conflicts can be resolved online but others can’t. Does the API give info about this?

Reply:
I have only found something called `potentialMergeCommit` with description:

```
The commit that GitHub automatically generated to test if this pull request could be merged. This field will not return a value if the pull request is merged, or if the test merge commit is still being generated. See the mergeable field for more details on the mergeability of the pull request.
```

I have no idea if this is something that we are looking for or not. It would be cool if we had a repo to test this feature.

Reply:
Okay. So this thing is something completely different, but there is another way that we can hack around to get the information if you can resolve conflicts on GitHub or not.
- Make a request to 
`https://github.com/owner/name/pulls/prNumber/conflicts`.
- Unauthorized
	- If the user hasn't got permissions to resolve conflicts we get `404` status.
- Authorized
	- If the user can resolve conflicts on GH we get `200` status without redirection - `{ redirected: false }`
	- If the option of resolving conflicts is not available for the PR we get `200` status with redirection - `{ redirected: true }`

This solution might be too volatile, but I haven't found anything better in Github API.

cc @fregante

Reply:
> It would be cool if we had a repo to test this feature.

Here's a PR list that has:

- PR without conflicts
- PR with resolvable conflicts (on GitHub.com)
- PR with unresolvable conflicts

https://github.com/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+author%3Afregante+created%3A%3C2018-01-31+

Reply:
>  * Make a request to
>    `https://github.com/owner/name/pulls/prNumber/conflicts`

I'd rather not make extra requests, on some pages there could be up to 20 of these requests. If this information is not available in the original API call, I'd just turn the icon into a link to the "mergeability details box"